com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -3438,17 +3438,23 @@ internal enum TransformExportType { Local, Global, Reset };
             Dictionary<GameObject, IExportData> exportData)
         {
             exportCancelled = false;
+            
+            m_lastFilePath = LastFilePath;
 
             // Export first to a temporary file
             // in case the export is cancelled.
             // This way we won't overwrite existing files.
-            try{
-                m_tempFilePath = Path.GetTempFileName();
+            try
+            {
+                // create a temp file in the same directory where the fbx will be exported
+                var exportDir = Path.GetDirectoryName(m_lastFilePath);
+                var tempFileName = Path.GetFileNameWithoutExtension(Path.GetRandomFileName());
+                // have to add .fbx to the end or else the temp file path won't match the file name
+                m_tempFilePath = exportDir + "/" + tempFileName + ".fbx";
```

Comment:
Here you should use `Path.Combine`. This will play better with other platforms

Reply:
I would also add the actual file name in m_tempFilePath. The metadata will at least give a hint about the right file name.

For example:
```
fileName = Path.GetFileName(m_lastFilePath);
m_tempFilePath = Path.Combine(exportDir, tempFileName + "_" + fileName);
```